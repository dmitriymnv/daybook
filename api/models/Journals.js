const {
  journals,
  journalsOffset,
  countJournals
} = require('../common/constants');

function queryJournals(body, res) {
  const { from, categories, publishers } = body.data;
  console.log('TCL: queryJournals -> publishers', publishers);
  if (from == 0) {
    queryJournalsFirst(categories, publishers, res);
  } else {
    queryJournalsFrom(categories, publishers, from, res);
  }
}

function queryJournalsFirst(categories, publishers, res) {
  database.query(countJournals(categories, publishers), (err, result) => {
    if (err) {
      dbQuery(res);
    } else {
      let { count } = result[0];
      database.query(journals(categories, publishers), (err, result) => {
        if (err) {
          dbQuery(res);
        } else {
          res.json({ count, result });
        }
      });
    }
  });
}

function queryJournalsFrom(categories, publishers, from, res) {
  database.query(
    journalsOffset(categories, from, publishers),
    (err, result) => {
      if (err) {
        dbQuery(res);
      } else {
        res.json({ result });
      }
    }
  );
}

module.exports = { queryJournals };
