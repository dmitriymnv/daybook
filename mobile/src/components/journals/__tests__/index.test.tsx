import React from 'react';
import renderer from 'react-test-renderer';

import Journals from '../index';

jest.mock('ScrollView', () => {
  const RealComponent = require.requireActual('ScrollView');
  const React = require('React');
  class ScrollView extends React.Component {
    scrollTo() {}

    render() {
      return React.createElement('ScrollView', this.props, this.props.children);
    }
  }
  ScrollView.propTypes = RealComponent.propTypes;
  return ScrollView;
});

jest.mock('axios', () => {
  // const data = [{ title: 'test article', publisher: 'test url' }];
  const data = {
    result: [
      { title: 'Заголовок1', pubsliher: 'Издатель1' },
      { title: 'Заголовок2', pubsliher: 'Издатель2' }
    ]
  };
  return {
    post: jest.fn(() => {
      return Promise.resolve({ data });
    })
  };
});

describe('Вывод журналов', () => {
  it('Возврат журналов при удачной загрузки', () => {
    const snap = renderer.create(<Journals />).root;

    expect(snap).toMatchSnapshot();
  });
});
