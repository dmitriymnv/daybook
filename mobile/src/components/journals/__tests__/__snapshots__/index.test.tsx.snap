// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Вывод журналов Возврат журналов при удачной загрузки 1`] = `
<View>
  <Text>
    Загрузка...
  </Text>
</View>
`;
